solidity/src/BaseRouter.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.25;
+
+import { TypeCasts } from "@hyperlane-xyz/libs/TypeCasts.sol";
+import { Router } from "@hyperlane-xyz/client/Router.sol";
+
+abstract contract BaseRouter is Router {
```

Comment:
What does this do beyond the Router, just expose the internal functions?

Reply:
- implements a `quoteGasPaymen`
- has a function for enrolling routers with the same address a the contract on receiving only the domains as param

Reply:
1. interesting, this reminds me that we probably want to inherit from GasRouter, not just router.
2. I would probably prefer to not add these to avoid bloating the bytecode. We originally started with using counterfactual addresses for routers, but it's rather hard to keep them consistent